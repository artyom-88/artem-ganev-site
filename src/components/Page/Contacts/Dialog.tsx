import React, { ReactNode } from 'react';
import './Dialog.scss';

interface IState {
    text: string;
}

interface IProps {
    closeHandler: () => void;
}

/**
 * Contact dialog component
 */
export default class Dialog extends React.Component<IProps, IState> {
    constructor(props: IProps) {
        super(props);
        this.state = { text: '' };
    }

    public render(): ReactNode {
        const { closeHandler } = this.props;
        const { text } = this.state;
        return (
            <div className='page-blog__root'>
                <div className='flexBox flexColumn alignItemsCenter justifyContentCenter page-blog__content'>
                    <form className='page-blog__form' onSubmit={ this.onSubmit }>
                        <input
                            className='page-blog__formInput'
                            type="text"
                            value={ text }
                            onChange={ this.onTextChange }
                            required={ true }
                        />
                        <button type="submit">Send</button>
                    </form>
                    <div className='page-blog__closeBtn'>
                        <button onClick={ closeHandler }>Close</button>
                    </div>
                </div>
            </div>
        );
    }

    private onTextChange = ({ target }: React.ChangeEvent<HTMLInputElement>) => {
        this.setState({ text: target.value });
    };

    private onSubmit = () => {
        // TODO:
        this.setState({ text: '' });
    };
}