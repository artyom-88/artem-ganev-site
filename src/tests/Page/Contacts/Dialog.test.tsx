import React from 'react';
import renderer from 'react-test-renderer';
import Dialog from '../../../components/Page/Contacts/Dialog';

test('Dialog', () => {
    const handler = () => {
        // Test if close click have happened
    };
    const component = renderer.create(<Dialog closeHandler={ handler }/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});