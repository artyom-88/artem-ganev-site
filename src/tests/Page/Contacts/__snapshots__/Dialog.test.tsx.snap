// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog 1`] = `
<div
  className="page-blog__root"
>
  <div
    className="flexBox flexColumn alignItemsCenter justifyContentCenter page-blog__content"
  >
    <form
      className="page-blog__form"
      onSubmit={[Function]}
    >
      <input
        className="page-blog__formInput"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Send
      </button>
    </form>
    <div
      className="page-blog__closeBtn"
    >
      <button
        onClick={[Function]}
      >
        Close
      </button>
    </div>
  </div>
</div>
`;
